core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockMetaStoreMetaOnly.java

```diff
@@ -35,6 +36,7 @@ public class RocksBlockMetaStoreMetaOnly extends RocksBlockMetaStore {
   // Map from block id to block locations.
   public final TwoKeyConcurrentMap<Long, Long, BlockLocation, Map<Long, BlockLocation>>
       mBlockLocations = new TwoKeyConcurrentMap<>(() -> new HashMap<>(4));
+  private final Map<BlockLocation, BlockLocation> mLocationCacheMap = new ConcurrentHashMap<>();
```

Comment:
This is a very useful change! Let's separate it into a different PR. Thanks a lot in advance!

Also I think this applies to all BlockMetaStore. The change should be in https://github.com/Alluxio/alluxio/blob/0f4e5978d3267d5fa6f81a07396477202a347417/core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java#L926 So when we need a BlockLocation, we always go to a cache instead of creating a new one.

Reply:
<img width="1343" alt="image" src="https://user-images.githubusercontent.com/17329931/212218644-d354efdb-10b3-4617-9847-227103d20455.png">

In fact we have two caller as above figure.

Reply:
@jiacheliu3 Create a PR https://github.com/Alluxio/alluxio/pull/16763 and move the cache logic to DefaultBlockMaster